staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go

```diff
@@ -352,8 +371,9 @@ func (w Wait) IsDeleted(event watch.Event) (bool, error) {
 
 // ConditionalWait hold information to check an API status condition
 type ConditionalWait struct {
-	conditionName   string
-	conditionStatus string
+	// conditionName   string
+	// conditionStatus string
+	conditions map[string]string
```

Comment:
Do we need a map here, or can we have a list of objects with explicit `name` and `status`

Reply:
I think it's much better to have a map and treat each condition as a single unit. This way we can just pass in the single struct to other methods.

Reply:
I meant, you can replace `conditions map[string]string` by 
```
type Condition struct {
  Name string
  Status string 
}
```

How do you know that people won't want to use the same condition name twice, which wouldn't work with the map?